extensions/wikia/WikiFactory/DeadWikis/maintenance.php

```diff
@@ -382,6 +398,86 @@ private function disableDiscussion( $cityId ) {
 		}
 	}
 
+	// Gets the most recent post from the specified site and returns whether it was made less than 60 days ago
+	private function isActiveSite($siteId)
+	{
+		$response = $this->getMostRecentPostForSite($siteId);
+
+		if ($response['postCount'] == 0) {
+			// no posts
+			return false;
+		}
+
+		// extract post creation date from response
+		$mostRecentPostCreationDate = $response['_embedded']['doc:posts'][0]['creationDate']['epochSecond'];
+
+		$sixtyDaysAgo = time() - 60*60*24*60;
+
+		return $mostRecentPostCreationDate > $sixtyDaysAgo;
+	}
+
+	private function getMostRecentPostForSite($siteId) {
+		$apiClient = $this->getSitesApi()->getApiClient();
+
+		$resourcePathTemplate = "/{siteId}/posts";
+		$httpBody = '';
+
+		// header params
+		$headerParams = ['Content-Type' => $apiClient->selectHeaderContentType(array('application/json'))];
+
+		$headerAccept = $apiClient->selectHeaderAccept(array('application/hal+json'));
+		if (!is_null($headerAccept)) {
+			$headerParams['Accept'] = $headerAccept;
+		}
+
+		// path params
+		$resourcePath = str_replace(
+				"{siteId}",
+				$apiClient->getSerializer()->toPathValue($siteId),
+				$resourcePathTemplate
+		);
+
+		// only need most recent post
+		$queryParams = [
+				'limit' => 1
+		];
+
+		// make the API Call
+		try {
+			list($rawResponse, $statusCode, $httpHeader) = $apiClient->callApi(
+					$resourcePath,
+					'GET',
+					$queryParams,
+					$httpBody,
+					$headerParams,
+					'object',
+					$resourcePathTemplate
+			);
+			$response = $apiClient->getSerializer()->deserialize($rawResponse, 'object', $httpHeader);
+		} catch (\Swagger\Client\ApiException $e) {
+			throw $this->processApiException($e, $apiClient);
+		}
+
+		return $response;
+	}
+
+	private function processApiException($e, $apiClient) {
+		switch ($e->getCode()) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a "case default" clause to this "switch" statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS131)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Replace this "switch" statement with "if" statements to increase readability. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1301)